compact.go

```diff
@@ -87,10 +95,13 @@ func newCompactorMetrics(r prometheus.Registerer) *compactorMetrics {
 }
 
 type compactorOptions struct {
-	maxBlockRange uint64
+	blockRanges []int64
 }
 
 func newCompactor(dir string, r prometheus.Registerer, l log.Logger, opts *compactorOptions) *compactor {
+	if opts.blockRanges == nil {
+		opts.blockRanges = defaultBlockRanges
+	}
```

Comment:
Do we need sanity checks on startup that the ranges actually fit within each other?
What happens if the chosen ranges do not work with already existing blocks?